message/manager.php

```diff
@@ -314,18 +314,18 @@ public function move_message($cannedmessage, $direction)
 		$sql = 'SELECT cannedmessage_id, cannedmessage_name, left_id, right_id
 			FROM ' . $this->cannedmessages_table . "
 			WHERE parent_id = {$cannedmessage['parent_id']}
-				AND " . (($direction == 'move_up') ? "right_id < {$cannedmessage['right_id']} ORDER BY right_id DESC" : "left_id > {$cannedmessage['left_id']} ORDER BY left_id ASC");
+				AND " . (($direction === 'move_up') ? "right_id < {$cannedmessage['right_id']} ORDER BY right_id DESC" : "left_id > {$cannedmessage['left_id']} ORDER BY left_id ASC");
 		$result = $this->db->sql_query_limit($sql, 1);
 		$target = $this->db->sql_fetchrow($result);
 		$this->db->sql_freeresult($result);
 
-		if (!sizeof($target))
+		if (!count($target))
 		{
 			// The canned message is already on top or bottom
 			return false;
 		}
 
-		if ($direction == 'move_up')
+		if ($direction === 'move_up')
 		{
 			$left_id = $target['left_id'];
 			$right_id = $cannedmessage['right_id'];
```

Comment:
Can you change [line 348](https://github.com/phpbb-extensions/cannedmessages/pull/1/files#diff-bcd3c83d9f08eff199a3efc2dac28aceR348) to:
```
$move_up_right = $target['right_id'];
```
Seems to be why moving messages down doesn't work.